inkcut/device/dialogs.enaml

```diff
@@ -110,13 +111,11 @@ enamldef DeviceDialog(Dialog): dialog:
                                 selected ::
                                     declaration = change['value']
                                     if declaration is not None:
-                                        transport = declaration.factory()
-                                        transport.declaration = declaration
+                                        protocol = None
                                         if device.connection and device.connection.protocol:
                                             # Copy from old connection
-                                            transport.protocol = device.connection.protocol
-
-                                        device.connection = transport
+                                            protocol = device.connection.protocol
```

Comment:
I guess we'd only want to do this when the 'old' protocol is also valid for the 'new' device - but OK

Reply:
Hmm seems to work fine though

Reply:
Whenever the device is changed it will reset both the connection and protocol to the defaults for that device. 

When the connection/transport is changed (what this handler is for) that should be independent from the protocol so we can simply reuse it.